x-pack/plugins/security_solution/public/common/links/links.ts

```diff
@@ -178,9 +178,16 @@ const getNormalizedLink = (id: SecurityPageName): Readonly<NormalizedLink> | und
 
 const processAppLinks = (appLinks: AppLinkItems, linksPermissions: LinksPermissions): LinkItem[] =>
   appLinks.reduce<LinkItem[]>((acc, { links, ...appLinkWithoutSublinks }) => {
+    // license check
     if (!isLinkAllowed(appLinkWithoutSublinks, linksPermissions)) {
+      if (linksPermissions.upselling.isPageUpsellable(appLinkWithoutSublinks.id)) {
+        acc.push({ ...appLinkWithoutSublinks, unauthorized: true });
+      }
+
```

Comment:
The `isLinkAllowed` function does not only check the license, it also checks the link experimental keys. 
We need to split this function into `isLinkLicenseAllowed()` and `isLinkExperimentalKeyAllowed()`.
And do:
```
    if (!isLinkExperimentalKeyAllowed(appLinkWithoutSublinks, linksPermissions)) {
      return acc;
    }
    
    if (
      !hasCapabilities(linksPermissions.capabilities, appLinkWithoutSublinks.capabilities) ||
      !isLinkLicenseAllowed(appLinkWithoutSublinks, linksPermissions)
    ) {
      if (linksPermissions.upselling.isPageUpsellable(appLinkWithoutSublinks.id)) {
        acc.push({ ...appLinkWithoutSublinks, unauthorized: true });
      }
      return acc; // not adding sub-links for links that are not authorized
    }
```